### DISCUSSION TITLE
ScriptReference/AssetDatabase.CreateAsset

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#255] [Notes]
# ScriptReference/AssetDatabase.CreateAsset

https://docs.unity3d.com/ScriptReference/AssetDatabase.CreateAsset.html

**Comment by joonjoonjoon** [comment]
I forgot to add the filename to the end of CreateAsset and it just executes without throwing an error but also without creating a file.  
Hard to debug this way.

**Comment by radiatoryang** [comment]
If you want to create text or script files with a template, you can use undocumented internal `UnityEditor.CreateScriptAsset("Assets/YourTemplatesScript.cs.txt", "DefaultNewScriptName.cs");` via reflection...  

```csharp
private static MethodInfo createScriptMethod = typeof(ProjectWindowUtil)  
    .GetMethod("CreateScriptAsset", BindingFlags.Static | BindingFlags.NonPublic);  
  
static void CreateScriptAsset(string templatePath, string destName) {  
    createScriptMethod.Invoke(null, new object[] { templatePath, destName });  
}
```
for more details, see [this forum thread](https://forum.unity.com/threads/how-to-create-your-own-c-script-template.459977/).
